File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {
        int res=0;
        for(auto x:nums) res=res|x;
        return recurse(nums,0,res,0);
    }
    int recurse(vector<int> &nums,int idx,int x,int count)
    {
        if(idx==nums.size()) return count==x;
        return recurse(nums,idx+1,x,count)+recurse(nums,idx+1,x,count|nums[idx]);
    }
};

File: 2170-count-number-of-maximum-bitwise-or-subsets/2170-count-number-of-maximum-bitwise-or-subsets.py
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        maxor = 0
        for num in nums:
            maxor |= num
        return self.recurse(nums,0,0,maxor)
    
    def recurse(self, nums: List[int], idx: int, curr: int, total: int) -> int:
        if idx==len(nums):
            return total==curr
        return self.recurse(nums,idx+1,curr,total)+self.recurse(nums,idx+1,curr|nums[idx],total)
        